app/src/test/java/org/oppia/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -0,0 +1,1088 @@
+    produceAndViewNextHint(hintPosition = 3, submitAnswer = this::submitWrongAnswerToState2AndWait)
+  }
+
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  private fun produceAndViewSolution(
+    activityScenario: ActivityScenario<StateFragmentTestActivity>,
+    revealedHintCount: Int
+  ) {
+    submitWrongAnswerToState2AndWait()
+    openHintsAndSolutionsDialog()
+    pressRevealSolutionButton(revealedHintCount)
+    clickConfirmRevealSolutionButton(activityScenario)
+    closeHintsAndSolutionsDialog()
+  }
+
+  private fun clickConfirmRevealSolutionButton(
+    activityScenario: ActivityScenario<StateFragmentTestActivity>
+  ) {
+    // See https://github.com/robolectric/robolectric/issues/5158 for context. It seems Robolectric
+    // has some issues interacting with alert dialogs. In this case, it finds and presses the button
+    // with Espresso view actions, but that button click doesn't actually lead to the click listener
+    // being called.
+    activityScenario.onActivity { activity ->
+      val hintAndSolutionDialogFragment = activity.supportFragmentManager.findFragmentByTag(
+        TAG_HINTS_AND_SOLUTION_DIALOG
+      )
+      val revealSolutionDialogFragment =
+        hintAndSolutionDialogFragment?.childFragmentManager?.findFragmentByTag(
+          TAG_REVEAL_SOLUTION_DIALOG
+        ) as? DialogFragment
+      val positiveButton =
+        revealSolutionDialogFragment?.dialog
+          ?.findViewById<View>(android.R.id.button1)
+      assertThat(checkNotNull(positiveButton).performClick()).isTrue()
+    }
+  }
+
+  /**
+   * Appends the specified text to a view. This is needed because Robolectric doesn't seem to
+   * properly input digits for text views using 'android:digits'. See
+   * https://github.com/robolectric/robolectric/issues/5110 for specifics.
+   */
+  @InternalCoroutinesApi
+  @ExperimentalCoroutinesApi
+  private fun appendText(text: String): ViewAction {
+    return object : ViewAction {
+      override fun getDescription(): String {
+        return "appendText($text)"
+      }
+
+      override fun getConstraints(): Matcher<View> {
+        return allOf(isEnabled())
+      }
+
+      override fun perform(uiController: UiController?, view: View?) {
+        (view as? EditText)?.append(text)
+        testCoroutineDispatchers.runCurrent()
+      }
+    }
+  }
+
+  // TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.
+  @Singleton
+  @Component(modules = [
+    TestDispatcherModule::class, ApplicationModule::class, NetworkModule::class,
+    LoggerModule::class, ContinueModule::class, FractionInputModule::class,
+    ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+    NumberWithUnitsRuleModule::class, NumericInputRuleModule::class, TextInputRuleModule::class,
+    InteractionsModule::class, GcsResourceModule::class, GlideImageLoaderModule::class,
```

Comment:
I am unable to run this test class. It was giving below error
<img width="1633" alt="Screenshot 2020-06-10 at 1 44 49 PM" src="https://user-images.githubusercontent.com/16301028/84244238-fcfd8e00-ab20-11ea-81ba-3f89a29fe674.png">

I tried adding  `@DragDropSortInputRules` annotation in InteractionsModule `provideDragAndDropSortInputInteractionClassifier` method. Error was resolved but after running I got below failed testcases.

<img width="1556" alt="Screenshot 2020-06-10 at 1 43 24 PM" src="https://user-images.githubusercontent.com/16301028/84244449-4948ce00-ab21-11ea-958b-f23f60018acd.png">

Reply:
I suspect the test may be out of date from ``develop``. I'll need to update it and see if I missed an import from new classes being added.

Reply:
I can verify that I hit these issues locally. I think a PR that went in 2 weeks ago triggered all the tests to start breaking. I'm investigating what needs to be done to fix it.

Reply:
Okay, I think it's fixed now. PTAL.

Reply:
It is fixed. Thanks!